### PR TITLE
Implement a clear class hierarchy

### PR DESCRIPTION
**Pull request by schacht-certat**

### CODE REVIEW

**Review by wagner-certat — CHANGES_REQUESTED** [review]
I'd appreciate changelog entries for the changes (new file `intelmq/lib/datatypes.py`, new mixin, new class `ExpertBot`, parent classes for all bots changed, ...). Otherwise: :tada:

File: intelmq/lib/datatypes.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class BotType(str, Enum):
+    COLLECTOR   = "Collector"
+    PARSER      = "Parser"
+    EXPERT      = "Expert"
+    OUTPUT      = "Output"
+
+    def toJson(self):
+        return self.value
```
Comment (wagner-certat) [inline intelmq/lib/datatypes.py:16]: Doesn't that miss the quotes necessary for making it a JSON-string?

**Comment by wagner-certat** [comment]
Could you please also update the dev guide (sections Template & Mixins are affected as I see with a quick search)

File: intelmq/lib/datatypes.py (hunk 1 above)
Comment (schacht-certat) [reply intelmq/lib/datatypes.py:16]: `toJson` is used by the JSON dump method to serialize individual values. In the case of the `BotType` class the dump method does not know how to represent the class (i.e. when dumping a bots attributes). Given that `self.value` is a string, the JSON dump method know how to handle the rest

Comment (wagner-certat) [reply intelmq/lib/datatypes.py:16]: Thanks for the explanation!

**Review by wagner-certat — APPROVED** [review]

**Pull request merged**